api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/AbstractPluginService.java

```diff
@@ -1,99 +1,159 @@
+/*******************************************************************************
+ *  Copyright 2023 Paladin Cloud, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License.  You may obtain a copy
+ *  of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ ******************************************************************************/
 package com.tmobile.pacman.api.admin.repository.service;
 
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicSessionCredentials;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.amazonaws.services.secretsmanager.AWSSecretsManagerClientBuilder;
+import com.amazonaws.services.secretsmanager.model.CreateSecretRequest;
+import com.amazonaws.services.secretsmanager.model.CreateSecretResult;
+import com.amazonaws.services.secretsmanager.model.DeleteSecretRequest;
+import com.amazonaws.services.secretsmanager.model.DeleteSecretResult;
+import com.amazonaws.services.secretsmanager.model.ResourceExistsException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
 import com.tmobile.pacman.api.admin.common.AdminConstants;
+import com.tmobile.pacman.api.admin.domain.ActivityLogRequest;
 import com.tmobile.pacman.api.admin.domain.ConfigPropertyItem;
 import com.tmobile.pacman.api.admin.domain.ConfigPropertyRequest;
+import com.tmobile.pacman.api.admin.domain.PluginParameters;
 import com.tmobile.pacman.api.admin.domain.PluginResponse;
-import com.tmobile.pacman.api.admin.domain.RedHatPluginRequest;
+import com.tmobile.pacman.api.admin.exceptions.PacManException;
```

Comment:
Let's stop using any prefix with "pac" like pacman, pacbot, pac.... This was a T-Mobile project and now we evolving the code as Paladin Cloud. 

Please rename the class and if it's a new class, would could start placing it into its own Paladin cloud package. 

Also, I think we should also be make more friendly exception types that have a nesting.

For example pacmanexception really tells me little in the name of what it could be containing. On the other hand something like ApiException or AdminApiException will tell me at least which component of the system is having an issue.